replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/SyncHelper.java

```diff
@@ -214,12 +204,16 @@ private Filter buildFilter() {
       throw new ReplicationException("Error creating filter from cql: " + config.getFilter(), e);
     }
 
-    if (lastSuccessfulRun != null) {
-      long time = lastSuccessfulRun.getStartTime().getTime();
-      if (lastSuccessfulRun.getLastSuccess() != null) {
-        time = lastSuccessfulRun.getLastSuccess().getTime();
-      }
-      Date timeStamp = new Date(time - 1000);
+    long timeOfLastSuccessfulRun = 0;
+    try {
+      timeOfLastSuccessfulRun =
+          history.getByReplicatorId(config.getId()).getLastSuccess().getTime();
+    } catch (NotFoundException | NullPointerException e) {
+      // config was never run or never successful
+    }
+
+    if (timeOfLastSuccessfulRun > 0) {
+      Date timeStamp = new Date(timeOfLastSuccessfulRun - 1000);
```

Comment:
❓ Why `- 1000`? THe reason should be documented here for the next person looking at the code.

Reply:
my point 😉